lib/bech32/src/Codec/Binary/Bech32/Internal.hs

```diff
@@ -183,7 +223,8 @@ dataPartToWords = mapMaybe charToWord5 . T.unpack . dataPartToText
 -- | Represents the human-readable part of a Bech32 string, as defined here:
 --   https://git.io/fj8FS
 newtype HumanReadablePart = HumanReadablePart Text
-    deriving (Eq, Show)
+    deriving newtype (Eq, Monoid, Semigroup)
+    deriving stock Show
```

Comment:
I wonder if there's any difference between `deriving newtype Eq` and `deriving stock Eq`, since that's a bit the "promise" with newtypes ("no-runtime cost"). I am just thinking out loud though :+1:

Reply:
I'm also curious about this. :)